internal/pkg/otel/builder-config.yaml

```diff
@@ -0,0 +1,43 @@
+dist:
+  name: otelcol
+  description: Elastic Distribution of OpenTelemetry Collector
+  module: otel
+  output_path: .
+
+extensions:
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/extension/healthcheckextension v0.107.0
```

Comment:
Unfortunately, we can't manage versions in this file and sync them with a go.mod file, since the dependencies are defined in the elastic-agent go.mod located at the root of the repository. We could consider using Go workspaces and moving this code to `${REPO_ROOT/otel}`, for example, so it can be its own module with its own go.mod file. This approach would also clean up the otelcol dependencies in the root go.mod file.

Reply:
We have dependabot currently bumping up the OTel dependency versions in `go.mod`: https://github.com/elastic/elastic-agent/blob/main/.github/dependabot.yml.  Perhaps we can add a new step or job to https://github.com/elastic/elastic-agent/blob/main/.github/workflows/post-dependabot.yml to do some parsing of the `go.mod` file and sync the versions in this file with the parsed results?